clang/lib/AST/Type.cpp

```diff
@@ -410,6 +410,12 @@ VectorType::VectorType(TypeClass tc, QualType vecType, unsigned nElements,
   VectorTypeBits.NumElements = nElements;
 }
 
+bool Type::isPackedVectorBoolType(const ASTContext &ctx) const {
+  if (ctx.getLangOpts().HLSL)
```

Comment:
This might be a dumb suggestion, but is the a way to just check if the mem reprsentation is i32 or i1? HLSL is probably the only language mode that needs this distinction but it feel like this shouldn't have a lang opt toggle based on the function name.

Reply:
The code to state that a bool vector should be a vector of i32s isn't accessible here.

Reply:
I think it would make for slightly better abstraction to add a `hasPackedBoolVectors()` accessor to `LangOptions` that returns `!HLSL`, similar to how `allowArrayReturnTypes()` works.